core/src/main/scala/org/apache/spark/api/python/PythonRDD.scala

```diff
@@ -77,22 +77,30 @@ private[spark] case class PythonFunction(
     broadcastVars: JList[Broadcast[PythonBroadcast]],
     accumulator: Accumulator[JList[Array[Byte]]])
 
+
+object PythonRunner {
+  def apply(func: PythonFunction, bufferSize: Int, reuse_worker: Boolean): PythonRunner = {
+    new PythonRunner(Seq(Seq(func)), bufferSize, reuse_worker, false, Seq(1))
+  }
+}
+
 /**
- * A helper class to run Python UDFs in Spark.
+ * A helper class to run Python mapPartition/UDFs in Spark.
  */
 private[spark] class PythonRunner(
-    funcs: Seq[PythonFunction],
+    funcs: Seq[Seq[PythonFunction]],
```

Comment:
This type is a little strange, so do you mind adding a scaladoc comment to explain what the two levels of nesting correspond to?